src/providers/doppler.js

```diff
@@ -15,17 +15,22 @@ async function fetch() {
     const authHeader = `Basic ${encodedAuthData}`;
     https
       .get(
-        "https://api.doppler.com/v3/configs/config/secrets/download",
+        "https://api.doppler.com/v3/configs/config/secrets/download?format=json",
         {
           headers: {
             Authorization: authHeader,
           },
         },
         (res) => {
-          let secrets = "";
-          res.on("data", (data) => (secrets += data));
+          let payload = "";
+          res.on("data", (data) => (payload += data));
           res.on("end", () => {
-            resolve(JSON.parse(secrets));
+            if (res.statusCode === 200) {
+              resolve(JSON.parse(payload));
+            } else {
+              const error = JSON.parse(payload).messages.join(" ");
```

Comment:
Can we wrap this in a try-catch and return a generic error with the status code if it fails? If we're experiencing downtime or we hit another issue upstream, this might not be JSON.

Reply:
Good call!